bundle/config/paths/paths.go

```diff
@@ -6,9 +6,13 @@ import (
 )
 
 type Paths struct {
-	// ConfigFilePath holds the path to the configuration file that
-	// described the resource that this type is embedded in.
+	// Absolute path on the local file system to the configuration file that holds
+	// the definition of this resource.
 	ConfigFilePath string `json:"-" bundle:"readonly"`
+
+	// Relative path from the bundle root to the configuration file that holds
+	// the definition of this resource.
+	RelativePath string `json:"relative_path,omitempty" bundle:"readonly"`
```

Comment:
This makes the field settable under the job resource.

Reply:
The field it annotated as `readonly` so it's not settable in user config.

Reply:
Note: With the custom unmarshaller we can add validation that `readonly` fields are not set. Should I file a followup for this?

Reply:
@shreyas-goenka @pietern as a workaround to avoid this error now, you can implement MarshalJSON for Paths structure and remove `json` annotation so it's not automatically set

Reply:
I think the issue is we don't want customers to set this value. This showing up in the serialised job representation is not an issue IMO.

Reply:
With custom MarhsalJSON you can remove `json` annotation so it's not set by customers but still can be serialised as needed

Reply:
Let's do this in the custom unmarshaller. The advantage of doing so is we explictly error our instead of silently not loading the field since it's absent from the schema. 

IMO it's also probably the right place structurally, but please correct me if I am wrong about that.